### PR TITLE
LibWeb: Some missing edge visitation fixes

### PR DESCRIPTION
**Pull request by Hendiadyoin1**
Related to #15217

also has a compat fix for our version of the test262 runner

CC @Lubrsi @davidot

### CODE REVIEW

File: Tests/LibJS/test262-runner.cpp [hunk 1]
```diff
@@ -550,7 +550,7 @@ void __assertion_failed(char const* assertion)
     handle_failed_assert(assertion);
 }
 #else
-[[noreturn]] void __assert_fail(char const* assertion, char const* file, unsigned int line, char const* function)
+extern "C" __attribute__((__noreturn__)) void __assert_fail(char const* assertion, char const* file, unsigned int line, char const* function)
```
Comment (davidot) [inline Tests/LibJS/test262-runner.cpp:553]: Oh interesting, GCC complained when I didn't add `[[noreturn]]`

File: Userland/Libraries/LibWeb/DOM/Event.cpp [hunk 2]
```diff
@@ -41,6 +41,20 @@ Event::Event(HTML::Window& window, FlyString const& type, EventInit const& event
 {
 }
 
+void Event::visit_edges(Visitor& visitor)
+{
+    visitor.visit(m_target.ptr());
+    visitor.visit(m_related_target.ptr());
+    visitor.visit(m_current_target.ptr());
+    for (auto& it : m_path) {
```
Comment (davidot) [inline Userland/Libraries/LibWeb/DOM/Event.cpp:49]: Is the `touch_target_list` of PathEntry guaranteed to have its entries in `m_touch_target_list`?

I don't know, and if it's hard to figure out maybe just leaving a FIXME is good enough for now.

File: Userland/Libraries/LibWeb/DOM/Event.cpp [hunk 3]
```diff
@@ -41,6 +41,20 @@ Event::Event(HTML::Window& window, FlyString const& type, EventInit const& event
 {
 }
 
+void Event::visit_edges(Visitor& visitor)
+{
+    visitor.visit(m_target.ptr());
```
Comment (davidot) [inline Userland/Libraries/LibWeb/DOM/Event.cpp:46]: We need `Base::visit_edges(visitor)` here to make sure the properties of the base class are also visited.

**Review by davidot — CHANGES_REQUESTED** [review]
Very nice, thanks for checking all this.

Did you look through all the files in LibWeb/LibJS?
Otherwise maybe listing the directories/files you looked at is nice if someone else wants to check the rest.

I did find one problem with a base class not being visited.

File: Userland/Libraries/LibWeb/DOM/Event.cpp (hunk 2 above)
Comment (Hendiadyoin1) [reply Userland/Libraries/LibWeb/DOM/Event.cpp:49]: Oh I missed that one,
looking at `append_to_path`, it seems like it isnt guaranteed

File: Tests/LibJS/test262-runner.cpp (hunk 1 above)
Comment (davidot) [reply Tests/LibJS/test262-runner.cpp:553]: Yeah as long as the CI passes it should be fine

**Review by davidot — APPROVED** [review]

File: Userland/Libraries/LibWeb/HTML/Worker.cpp [hunk 4]
```diff
@@ -36,6 +36,9 @@ void Worker::visit_edges(Cell::Visitor& visitor)
     visitor.visit(m_document.ptr());
     visitor.visit(m_implicit_port.ptr());
     visitor.visit(m_outside_port.ptr());
+    visitor.visit(m_worker_realm.ptr());
+    visitor.visit(m_worker_scope.ptr());
+    visitor.visit(m_worker_window.ptr());
```
Comment (awesomekling) [inline Userland/Libraries/LibWeb/HTML/Worker.cpp:41]: These objects belong to a different GC heap in another entirely `JS::VM` (`m_worker_vm`). They should not be visited here.

**Review by awesomekling — CHANGES_REQUESTED** [review]

**Comment by Hendiadyoin1** [comment]
> Did you look through all the files in LibWeb/LibJS?

Only LibWeb with a search for GCPtr and the file match `LibWeb/*.h, LibWeb/*/*.h`

File: Userland/Libraries/LibWeb/HTML/Worker.cpp (hunk 4 above)
Comment (Hendiadyoin1) [reply Userland/Libraries/LibWeb/HTML/Worker.cpp:41]: Ah OK, did not see the note about that

**Pull request merged**